src/random_ray/flat_source_domain.cpp

```diff
@@ -184,9 +197,17 @@ void FlatSourceDomain::normalize_scalar_flux_and_volumes(
 void FlatSourceDomain::set_flux_to_flux_plus_source(
   int64_t sr, double volume, int g)
 {
-  double sigma_t = sigma_t_[source_regions_.material(sr) * negroups_ + g];
-  source_regions_.scalar_flux_new(sr, g) /= (sigma_t * volume);
-  source_regions_.scalar_flux_new(sr, g) += source_regions_.source(sr, g);
+  int material = source_regions_.material(sr);
+  if (material == MATERIAL_VOID) {
+    source_regions_.scalar_flux_new(sr, g) /= volume;
+    source_regions_.scalar_flux_new(sr, g) +=
+      0.5f * source_regions_.external_source(sr, g) *
```

Comment:
Why 0.5f when other terms in this operation are `double`?

Reply:
For this one, the `source_regions_.external_source(sr, g)` array is stored in FP32. The fluxes are done in FP64 as they potentially have hundreds or thousands of accumulations as an unbounded number of ray crossings can occur in a single iteration. The source though is just computed directly from the flux once per iteration, so it is much less sensitive to numerical precision as compared to the flux or volume.